src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -738,7 +741,7 @@ private async Task<ConnectBlocksResult> ConnectBlockAsync(ChainedHeaderBlock blo
             }
 
             // Call the validation engine.
-            ValidationContext validationContext = await this.consensusRules.FullValidationAsync(blockToConnect).ConfigureAwait(false);
+            ValidationContext validationContext = await this.fullValidator.ValidateAsync(blockToConnect.ChainedHeader, blockToConnect.Block).ConfigureAwait(false);
```

Comment:
Please change interface of the validator to accept `ChainedHeaderBlock`

Reply:
Do you mean the signature for `this.consensusRules.FullValidationAsync` (which has a different signature to the PV version for some reason)? All the validators in BlockValidator.cs currently have a `(header, block)` signature, unless you want that to be changed unilaterally.

Reply:
Yeah, change the signature. 
It looks like internally validator constructs ChainedHeaderBlock again so we might easily save allocation here by avoiding creating another ChainedHeaderBlock there.

Reply:
Okay, if it saves allocation then we should do it

Reply:
Why not instead pass `ChainedHeaderBlock` everywhere?

Reply:
I wouldn't mind either, but I have unsupported feeling that it would lead to more allocations

Reply:
Ok lets keep it like this for now its fine